runtime/compiler/env/DependencyTable.cpp

```diff
@@ -0,0 +1,232 @@
+/*******************************************************************************
+ * Copyright IBM Corp. and others 2024
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] https://openjdk.org/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+ *******************************************************************************/
+
+#if !defined(PERSISTENT_COLLECTIONS_UNSUPPORTED)
```

Comment:
Might be cleaner to do what is done for JITServer / CRIU, eg https://github.com/eclipse-openj9/openj9/blob/master/runtime/compiler/control/CMakeLists.txt#L36-L48

Reply:
Although I suppose the tradeoff is that you'd have to guard all places that include the header appropriately as well.

Reply:
I just pushed to include `DependencyTable.cpp` conditionally in the `CMakeLists.txt`, but I don't think the way I did it works. I think I'd need to define a feature flag so that `DependencyTable.cpp` is conditionally included in `common.mk.ftl` as well.

Reply:
I've reverted that.